Library/Homebrew/test/cmd/upgrade_spec.rb

```diff
@@ -17,4 +17,17 @@
     expect(HOMEBREW_CELLAR/"testball/0.1").to be_a_directory
     expect(HOMEBREW_CELLAR/"testball/0.0.1").not_to exist
   end
+
+  it "upgrades a Formula and cleans up old versions when `HOMEBREW_INSTALL_CLEANUP` is set" do
+    setup_test_formula "testball"
+    # allow(ENV).to receive(:[]).and_call_original
+    # allow(ENV).to receive(:[]).with("HOMEBREW_INSTALL_CLEANUP").and_return("1")
+    ENV["HOMEBREW_INSTALL_CLEANUP"] = "1"
```

Comment:
I've tried both manually setting into `ENV` and mocking it out as shown in the 2 commented lines. I wasn't able to get either working even though I could confirm the functionality manually.

Reply:
@Wojonatior Ideally we wouldn't add another integration test for this (they are really slow). Either add a unit test or omit one entirely.